ansible_builder/introspect.py

```diff
@@ -125,14 +163,76 @@ def get_dependency(self, entry):
         return req_file
 
 
+def simple_combine(reqs):
+    """Given a dictionary of requirement lines keyed off collections,
+    return a list with the most basic of de-duplication logic,
+    and comments indicating the sources based off the collection keys
+    """
+    consolidated = []
+    fancy_lines = []
+    for collection, lines in reqs.items():
+        for line in lines:
+            if line_is_empty(line):
+                continue
+
+            base_line = line.split('#')[0].strip()
+            if base_line in consolidated:
+                i = consolidated.index(base_line)
+                fancy_lines[i] += ', {}'.format(collection)
+            else:
+                fancy_line = base_line + '  # from collection {}'.format(collection)
+                consolidated.append(base_line)
+                fancy_lines.append(fancy_line)
+
+    return fancy_lines
+
+
+def write_files(data, write_pip=None, write_bindep=None):
+    if write_pip and data.get('python'):
+        with open(write_pip, 'w') as f:
+            f.write('\n'.join(simple_combine(data.get('python')) + ['']))
+    if write_bindep and data.get('system'):
+        with open(write_bindep, 'w') as f:
+            f.write('\n'.join(simple_combine(data.get('system')) + ['']))
+
+
+def parse_introspect_output(stdout):
+    p = re.compile(
+        '{0}(?P<yaml_text>.+){1}'.format(begin_delimiter, end_delimiter),
+        flags=re.MULTILINE | re.DOTALL
+    )
+    m = p.search(stdout)
+    if m is None:
+        return None
+    yaml_text = m.group('yaml_text').strip()
+    data = yaml.safe_load(yaml_text)
+    return data
+
+
 def add_introspect_options(parser):
     parser.add_argument(
-        'folders', default=[base_collections_path], nargs='*',
+        'folder', default=base_collections_path, nargs='?',
         help=(
             'Ansible collections path(s) to introspect. '
             'This should have a folder named ansible_collections inside of it.'
         )
     )
+    parser.add_argument(
+        '--user-pip', dest='user_pip',
+        help='An additional file to combine with collection pip requirements.'
+    )
+    parser.add_argument(
+        '--user-bindep', dest='user_bindep',
+        help='An additional file to combine with collection bindep requirements.'
+    )
+    parser.add_argument(
+        '--write-pip', dest='write_pip',
+        help='Write the combined bindep file to this location.'
+    )
+    parser.add_argument(
+        '--write-bindep', dest='write_bindep',
+        help='Write the combined bindep file to this location.'
+    )
```

Comment:
Not using `--write-pip`, but added it for consistency.

We have the introspect script write the bindep file because of technicalities of how bindep operates. The first phase needs to have written a file ready to pass to `bindep`